model/DataObject.php

```diff
@@ -1744,63 +1756,90 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
 	}
 
 	/**
+	 * @deprecated 4.0 Method has been replaced by hasOne() and hasOneComponent()
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function has_one($component = null) {
+		if($component) {
+			Deprecation::notice('4.0', 'Please use hasOneComponent() instead');
+			return $this->hasOne($component);
```

Comment:
Oops, I did that on the other methods, just missed this one! I can wrap the rest of the function in an else if you like, personal preference I guess :)